src/test/spec/retryable_reads.rs

```diff
@@ -0,0 +1,174 @@
+                options.retry_reads = Some(client_options.get_bool("retryReads").unwrap());
+            }
+            if TestClient::new().await.is_sharded() && test_case.use_multiple_mongoses != Some(true)
+            {
+                options.hosts = options.hosts.iter().cloned().take(1).collect();
+            }
+            let client = EventClient::with_options(options).await;
+
+            if let Some(ref run_on) = test_file.run_on {
+                let can_run_on = run_on.iter().any(|run_on| run_on.can_run_on(&client));
+                if !can_run_on {
+                    println!("Skipping {}", test_case.description);
+                    continue;
+                }
+            }
+
+            let db_name = match test_file.database_name {
+                Some(ref db_name) => db_name.clone(),
+                None => test_case.description.replace('$', "%").replace(' ', "_"),
+            };
+
+            let coll_name = match test_file.collection_name {
+                Some(ref coll_name) => coll_name.clone(),
+                None => "coll".to_string(),
+            };
+
+            let coll = client.init_db_and_coll(&db_name, &coll_name).await;
+
+            if let Some(ref data) = test_file.data {
+                match data {
+                    TestData::Single(data) => {
+                        if !data.is_empty() {
+                            coll.insert_many(data.clone(), None)
+                                .await
+                                .expect(&test_case.description);
+                        }
+                    }
+                    TestData::Many(_) => panic!("{}: invalid data format", &test_case.description),
+                }
+            }
+
+            if let Some(ref fail_point) = test_case.fail_point {
+                client
+                    .database("admin")
+                    .run_command(fail_point.clone(), None)
+                    .await
+                    .unwrap();
+            }
+
+            let mut events: Vec<TestEvent> = Vec::new();
+            for operation in test_case.operations {
+                let test_operation: AnyTestOperation =
+                    bson::from_bson(Bson::Document(operation.as_document())).unwrap();
+                let result = match operation.object {
+                    OperationObject::Client => client.run_client_operation(&test_operation).await,
+                    OperationObject::Database => {
+                        client
+                            .run_database_operation(&test_operation, &db_name)
+                            .await
+                    }
+                    OperationObject::Collection => {
+                        client
+                            .run_collection_operation(&test_operation, &db_name, &coll_name)
+                            .await
+                    }
+                    OperationObject::GridfsBucket => panic!("unsupported operation type"),
```

Comment:
Should we just skip this (i.e. `continue` to the next iteration) instead of erroring?

Reply:
This branch should never be hit since the runner should `continue` if it finds "gridfs" in the operation names at the beginning of the function. If it were to execute it would indicate that something was going wrong in the runner, so I think it's more helpful to have it panic instead of fail silently.

Reply:
Makes sense. I think it would be useful for the panic message to mention the operation name in this case.